karmagrambot/analytics.py

```diff
@@ -74,10 +79,34 @@ def get_top_n_karmas(chat_id: int, n: int) -> List[UserKarma]:
 
     return top_n
 
+def get_devil_saint(chat_id: int) -> DevilSaint:
+    """Get the users with the lowest and the greatest karma in a given chat.
+
+    Args:
+        chat_id: The id of the chat that we'll be searching in.
+
+    Returns:
+        A dictionary with the informations of both devil and saint.
+    """
+
+    db = dataset.connect(DB_URI)
+    users = db['tracked'].find(chat_id=chat_id)
+    user_ids = [u['user_id'] for u in users]
+
+    devil_id = sorted(user_ids, key=lambda u: get_karma(u, chat_id))[0]
+    saint_id = user_ids[-1]
```

Comment:
You are not accessing the sorted version of the id list here. This should look like

```
sorted_ids = sorted(user_ids, key=lambda u: get_karma(u, chat_id))
devil_id = sorted_ids[0]
saint_id = sorted_ids[-1]
```